core/trino-main/src/main/java/io/trino/execution/QueryManagerConfig.java

```diff
@@ -298,6 +299,20 @@ public QueryManagerConfig setQueryExecutorPoolSize(int queryExecutorPoolSize)
         return this;
     }
 
+    @Min(1)
+    public int getMaxStateMachineThreadsPerQuery()
+    {
+        return maxStateMachineThreadsPerQuery;
+    }
+
+    @Config("query.max-state-machine-threads-per-query")
```

Comment:
"per-query" suffix seems redundant, maybe rename it to `query.state-machine-max-callback-threads` (e.g. `query.remote-task.max-callback-threads`).

Reply:
In this case it's not redundant: the limit is enforced on a per-query basis instead of being enforced by determining the global configuration of the the shared threadpool executor. Instead, each `QueryStateMachine` gets it's own `BoundedExecutor` to enforce the limit independent of one another.

Reply:
It appears that the rest of the configs in this file which are enforced on a per-query basis don't add the "per-query" suffix. E.g. we use query.max-stage-count and query.stage-count-warning-threshold, not query.max-stage-count-per-query and query.stage-count-warning-threshold-per-query.

Reply:
Fair enough, I've changed the config property name to `query.max-state-machine-callback-threads` in the latest revision.